client/dashboard/app/queries/domain-dns-records.ts

```diff
@@ -0,0 +1,8 @@
+import { queryOptions } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+export const domainDnsRecordsQuery = ( domain: string ) =>
+	queryOptions( {
+		queryKey: [ 'domain', domain, 'dns-records' ],
+		queryFn: () => wpcom.req.get( `/domains/${ domain }/dns` ),
```

Comment:
The GET request could be moved to a data layer as you did for the **addDNSRecord**.

Reply:
Oops, this query wasn't actually part of this PR 😄 I'll remove it